zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionBehavior.java

```diff
@@ -187,6 +191,23 @@ void distributeNextInQueue(final long finishedDistributionKey, final int partiti
                     nextDistributionKey));
   }
 
+  private void continueAfterQueue(final String queue) {
+    if (distributionState.hasQueuedDistributions(queue)) {
+      return;
+    }
+    distributionState.forEachContinuationCommand(queue, this::handleContinuationCommand);
+  }
+
+  private void handleContinuationCommand(final long key, final CommandDistributionRecord record) {
+    final var intent = record.getIntent();
+    final var value = record.getCommandValue();
+
+    commandWriter.appendFollowUpCommand(key, intent, value);
+
+    record.setPartitionId(currentPartitionId);
+    stateWriter.appendFollowUpEvent(key, CommandDistributionIntent.CONTINUATION_COMPLETED, record);
```

Comment:
❌ We shouldn't use the full record here since we can easily run out of the maximum batch size this way. As you only need the key and queueId we should create an empty record with only those properties (and other non-optional properties) set.

Reply:
Fixed in 950df4b